judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/hibernate/ContestHibernateDao.java

```diff
@@ -79,4 +96,14 @@ public ContestHibernateDao(SessionFactory sessionFactory, Clock clock, ActorProv
             return cb.greaterThanOrEqualTo(endTime, cb.literal(currentInstantEpoch));
         };
     }
+
+    static void applySearchOptions(FilterOptions.Builder<ContestModel> filterOptions, SearchOptions searchOptions) {
+        if (searchOptions.getTerms().containsKey("name")) {
+            filterOptions.putColumnsLike(ContestModel_.name, contains(searchOptions.getTerms().get("name")));
+        }
+    }
+
+    static String contains(String str) {
```

Comment:
I think let's move this logic into `UnmodifiableHibernateDao.java::appyFilters()` itself. Let's assume that all search term will have `%term%` LIKE-expression.